### PR TITLE
Fixes the electrified arm for implants

### PR DESCRIPTION
**Pull request by MarsM0nd**
The electrified arm, if used by humans, currently runtime, as they search for a borg cell on a human. Adds a subtype that does have it's own power supply, that recharges over time. The recharge should be slow enough to stop it from being used to permastun someone. The speed the recharge does happen can be easily edited for adminbuse, or changed if it needs to be re-balanced.
:cl:
Fix: Stops the electrified arm from combat and stunbaton implant from runtiming.
Add: Adds a self-recharging battery to the electrified arm implant, limiting the uses you have in a given timeframe.
/:cl:

### CODE REVIEW

**Comment by shazbot194** [comment]
You could make it use nutrition and fail when there isn't x amount of nutrition

**Comment by taukausanake** [comment]
I'd rather have it require an internal battery on the implant, but that might be more trouble than it's worth. Not sure how best to recharge it too in that case

**Comment by KingPhilipIII** [comment]
I recently discovered emagged toolset arm implants give you a cyborg knife which basically does a ton of damage while being impossible to disarm from you. Nice stuff.

**Comment by MarsM0nd** [comment]
Added an internal battery for the implant, which gets recharged over time. Should now let you stun someone 10 times in a row without the recharge proccing, which is the same ammount of charge a standart baton does have. Unless i'm mistaken, you should recharge 5 uses in one minute, as 6 uses in a minute could mean you can permastun a person with that arm, given that `stun(5)` should stun one for 5 processing ticks, which are 10 seconds. Updating OP to reflect the change.

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -15,7 +15,9 @@
 			playsound(M, 'sound/weapons/Genhit.ogg', 50, 1)
 			return 0
 
-	if(!user.cell.use(charge_cost))
```
Comment (Fox-McCloud) [inline code/game/objects/items/robot/robot_items.dm]: just make this

```
	if(iscyborg(user))
		var/mob/living/silicon/robot/R = user
		if(!R.cell.use(charge_cost))
			return
``` 

and be done with it.

**Comment by MarsM0nd** [comment]
Made the alternative with the sanity check only, as while the arm should probably have some sort of restriction on its use, it might be done better than i have done it, and as it's currently admin spawnable only, making it functional should be prioritized.

**Comment by tigercat2000** [comment]
Made redundant by #8156

**Pull request closed without merging**